Assets/_PlatformerDevelopment/Scripts/Enemy/Enemy.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using UnityEngine;
+
+namespace PersonalDevelopment
+{
+    public enum EnemyState
+    {
+        Patrol,
+        Attack
+    }
+    
+    [RequireComponent(typeof(BoxCollider))]
+    [RequireComponent(typeof(Rigidbody))]
+    public abstract class Enemy : MonoBehaviour
+    {
+        protected BoxCollider _detectionCollider = null;
+        protected Rigidbody _rigidBody = null;
+        
+        protected EnemyState _state = EnemyState.Patrol;
+        protected bool _canAttack = false;
+        protected GameObject _player;
+
+        [SerializeField] private float _moveDistance = 5f;
+        [SerializeField] private float _moveSpeed = 5f;
+        [SerializeField] private bool _willMoveLeftFirst = false;
+
+        private BotMovement _botMovement = null;
+        
+        protected virtual void Setup()
+        {
+            _detectionCollider = GetComponent<BoxCollider>();
+            _rigidBody = GetComponentInParent<Rigidbody>();
+            _state = EnemyState.Patrol;
+        }
+
+        protected abstract void Attack();
+
+        protected virtual void Patrol()
+        {
+            Vector3 destination = _botMovement.GetDestination(transform.position, _moveSpeed);
```

Comment:
You shouldn't put the Patrol here. What this should do. Is grab the BotMovement script GetComponent<BotMovement>(); and then call a function from there.

Reply:
Reason why it's only moving one way, is because the botMovement is called twice.